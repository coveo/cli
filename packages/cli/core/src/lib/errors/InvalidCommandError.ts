import {CLIBaseError} from '@coveo/cli-commons/errors/cliBaseError';

export class InvalidCommandError extends CLIBaseError {
  public name = 'Invalid Command Error';
  public constructor(reason: string) {
    super(reason);
  }
}
