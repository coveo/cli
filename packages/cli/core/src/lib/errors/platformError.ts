import {PlatformPrivilege} from '@coveo/cli-commons/preconditions/platformPrivilege';
import {
  PrintableError,
  SeverityLevel,
} from '@coveo/cli-commons/errors/printableError';

export class MissingPrivilegeError extends PrintableError {
  public constructor(
    public privilege: PlatformPrivilege,
    public anonymous?: boolean
  ) {
    super(SeverityLevel.Error);
    this.message = privilege.unsatisfiedConditionMessage(Boolean(anonymous));
  }
}
