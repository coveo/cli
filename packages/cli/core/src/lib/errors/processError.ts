import {
  PrintableError,
  SeverityLevel,
} from '@coveo/cli-commons/src/errors/printableError';

export class ProcessAbort extends PrintableError {
  public name = 'Operation Aborted';
  public constructor(message = 'Operation aborted') {
    super(SeverityLevel.Info);
    this.message = message;
  }
}
