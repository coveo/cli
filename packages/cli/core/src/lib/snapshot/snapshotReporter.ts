import {
  ResourceSnapshotsReportModel,
  ResourceSnapshotsReportOperationModel,
  ResourceSnapshotsReportOperationResult,
  ResourceSnapshotType,
} from '@coveo/platform-client';
import {
  ReportViewerOperationName,
  ReportViewerResourceReportModel,
  SnapshotReportStatus,
} from './reportPreviewer/reportPreviewerDataModels';

type ResourceEntries = [string, ResourceSnapshotsReportOperationModel];

type NoopHandler = (this: void) => void;
type SnapshotReporterHandler = (this: SnapshotReporter) => void | Promise<void>;

type SnapshotReporterHandlers = {
  [K in SnapshotReportStatus]: SnapshotReporterHandler | NoopHandler;
};

const MissingVaultEntryResultCode = 'INVALID_PLACEHOLDER';
//#endregion

export type VaultEntryAttributes = {
  vaultEntryId: string;
  resourceName: string;
  resourceType: ResourceSnapshotType;
};

export class SnapshotReporter {
  private missingVaultEntriesSet: Set<VaultEntryAttributes> = new Set();

  private operationResultsParseFuseUsed: boolean = false;

  private consumeOperationResultsParseFuse() {
    if (this.operationResultsParseFuseUsed) {
      return false;
    } else {
      this.operationResultsParseFuseUsed = true;
      return true;
    }
  }

  public get missingVaultEntries() {
    return this.missingVaultEntriesSet.values();
  }
  public resourceInErrorCount: number = 0;
  public resourceInError: Map<Partial<ResourceSnapshotType>, Set<string>> =
    new Map();
  public constructor(public readonly report: ResourceSnapshotsReportModel) {}

  public static convertResourceEntriesToResourceSnapshotsReportOperationModel([
    name,
    operations,
  ]: ResourceEntries) {
    return {name, operations};
  }

  public static resourceHasAtLeastOneOperationFilter(
    operationsToDisplay: ReportViewerOperationName[]
  ) {
    return ([_, operations]: ResourceEntries) =>
      operationsToDisplay.reduce(
        (previous, current) => previous + operations[current],
        0
      ) > 0;
  }

  public getAffectedResourceTypes(): ResourceSnapshotType[] {
    const isResourceTypeReportAffected = (
      report: ResourceSnapshotsReportOperationModel
    ) => Object.values(report).some((operation) => operation > 0);
    const changedResourcesTypes: ResourceSnapshotType[] = [];
    for (const resourceType of Object.keys(this.report.resourceOperations)) {
      if (
        isResourceTypeReportAffected(
          this.report.resourceOperations[resourceType]
        )
      ) {
        changedResourcesTypes.push(resourceType as ResourceSnapshotType);
      }
    }
    return changedResourcesTypes;
  }

  public hasChangedResources() {
    const totalUnchanges =
      this.getOperationTypeTotalCount('resourcesUnchanged');
    return totalUnchanges !== this.report.resourcesProcessed;
  }

  public getOperationTypeTotalCount(
    type: keyof ResourceSnapshotsReportOperationModel
  ) {
    const count = Object.values(this.report.resourceOperations).reduce(
      (count: number, current: ResourceSnapshotsReportOperationModel) =>
        count + current[type],
      0
    );

    return count;
  }

  private reportHandlers: SnapshotReporterHandlers = {
    [SnapshotReportStatus.SUCCESS]: () => {},
    [SnapshotReportStatus.NO_CHANGES]: () => {},
    [SnapshotReportStatus.MISSING_VAULT_ENTRIES]: () => {},
    [SnapshotReportStatus.ERROR]: () => {},
  };

  public setReportHandler<T extends SnapshotReportStatus>(
    status: T,
    handler: SnapshotReporterHandlers[T]
  ): SnapshotReporter {
    this.reportHandlers[status] = handler;
    return this;
  }

  public async handleReport(): Promise<void> {
    const reportStatuses = this.getReportStatuses();
    await this.executeHandlers(reportStatuses.fixables);
    if (reportStatuses.errors.length > 0) {
      await this.executeHandlers(reportStatuses.errors);
      return;
    }
    if (reportStatuses.successes.length > 0) {
      await this.executeHandlers(reportStatuses.successes);
    }
  }

  private async executeHandlers(reportStatuses: SnapshotReportStatus[]) {
    for (const status of reportStatuses) {
      await this.callAndReset(status);
    }
  }

  private async callAndReset(handler: SnapshotReportStatus) {
    await this.reportHandlers[handler].apply(this);
    this.reportHandlers[handler] = () => {};
  }

  private getReportStatuses() {
    const statuses: Record<
      'fixables' | 'successes' | 'errors',
      SnapshotReportStatus[]
    > = {
      fixables: [],
      successes: [],
      errors: [],
    };

    this.parseResourcesOperationsResults();
    if (this.missingVaultEntriesSet.size > 0) {
      statuses.fixables.push(SnapshotReportStatus.MISSING_VAULT_ENTRIES);
    }

    if (!this.hasChangedResources()) {
      statuses.successes.push(SnapshotReportStatus.NO_CHANGES);
    }

    if (this.resourceInErrorCount > 0) {
      statuses.errors.push(SnapshotReportStatus.ERROR);
    } else {
      statuses.successes.push(SnapshotReportStatus.SUCCESS);
    }

    return statuses;
  }

  private parseResourcesOperationsResults(): void {
    if (!this.consumeOperationResultsParseFuse()) {
      return;
    }
    for (const [resourceType, resource] of Object.entries(
      this.report.resourceOperationResults
    )) {
      for (const [resourceName, errors] of Object.entries(resource)) {
        for (const err of errors) {
          this.parseResourceOperationResult(err, resourceName, resourceType);
        }
      }
    }
  }

  private parseResourceOperationResult(
    err: ResourceSnapshotsReportOperationResult,
    resourceName: string,
    resourceType: string
  ) {
    const missingEntry = SnapshotReporter.tryGetMissingVaultEntryName(err);
    if (missingEntry) {
      this.missingVaultEntriesSet.add({
        vaultEntryId: missingEntry,
        resourceName: resourceName,
        resourceType: resourceType as ResourceSnapshotType,
      });
    } else {
      // TODO: Fix PlatformClient to reflect proper typing.
      this.addResourceInError(resourceType as ResourceSnapshotType, err);
    }
  }

  private addResourceInError(
    resourceType: ResourceSnapshotType,
    err: ResourceSnapshotsReportOperationResult
  ) {
    this.resourceInErrorCount++;
    let errorSet = this.resourceInError.get(resourceType);
    if (!errorSet) {
      errorSet = new Set();
      this.resourceInError.set(resourceType, errorSet);
    }
    errorSet.add(`${err.resultCode}: ${err.message}`);
  }

  private static missingVaultEntryMatcher =
    /^The vault entry referenced by \{\{ VAULT\.(?<entryName>.*) \}\} could not be found in the vault\.$/;

  private static tryGetMissingVaultEntryName(
    err: ResourceSnapshotsReportOperationResult
  ): string | undefined {
    let message: string = '';
    if (typeof err !== 'string') {
      if (err.resultCode === MissingVaultEntryResultCode) {
        message = err.message;
      } else {
        return undefined;
      }
    } else {
      message = err;
    }
    const match = message.match(SnapshotReporter.missingVaultEntryMatcher);
    return match?.groups?.['entryName'];
  }

  public getChangedResources(
    operationsToDisplay: ReportViewerOperationName[]
  ): ReportViewerResourceReportModel[] {
    return Object.entries(this.report.resourceOperations)
      .filter(
        SnapshotReporter.resourceHasAtLeastOneOperationFilter(
          operationsToDisplay
        )
      )
      .map(
        SnapshotReporter.convertResourceEntriesToResourceSnapshotsReportOperationModel
      );
  }

  public get type() {
    const type = this.prettify(this.report.type);
    return type;
  }

  public get status() {
    const status = this.prettify(this.report.status);
    return status;
  }

  public get resultCode() {
    const status = this.prettify(this.report.resultCode);
    return status;
  }

  private prettify(str: string): string {
    const capitalized = (word: string) =>
      word.charAt(0) + word.slice(1).toLowerCase();

    const spaced = str.replace(/_/g, ' ');
    return spaced.split(' ').map(capitalized).join(' ');
  }
}
