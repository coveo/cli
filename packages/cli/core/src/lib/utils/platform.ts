import {Config} from '@coveo/cli-commons/config/config';

export function getTargetOrg(config: Config, target?: string) {
  if (target) {
    return target;
  }
  const cfg = config.get();
  return cfg.organization;
}
