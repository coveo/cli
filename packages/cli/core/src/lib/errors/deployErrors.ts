import {CLIBaseError} from '@coveo/cli-commons/errors/cliBaseError';
import {getPrettyJsonValidationErrors} from '@coveo/cli-commons/errors/jsonError';
import {ValidationError} from 'jsonschema';

export class DeployConfigError extends CLIBaseError {
  public name = 'Invalid Deploy Config Error';
  public constructor(filePath: string, errors: ValidationError[]) {
    super(`\nParsing: ${filePath}.${getPrettyJsonValidationErrors(errors)}`);
  }
}
