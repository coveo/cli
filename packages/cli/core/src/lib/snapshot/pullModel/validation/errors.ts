import {ValidationError} from 'jsonschema';
import {CLIBaseError} from '@coveo/cli-commons/errors/cliBaseError';
import {getPrettyJsonValidationErrors} from '@coveo/cli-commons/errors/jsonError';

abstract class BaseSPMError extends CLIBaseError {
  public name = 'Snapshot Pull Model Error';
  private static messageSuffix =
    '\nThis is probably a problem with the Coveo CLI, please report this issue at https://github.com/coveo/cli/issues';
  public constructor(message: string, shouldContactCoveo: boolean) {
    super(`${message}${shouldContactCoveo ? BaseSPMError.messageSuffix : ''}`);
  }
}

export class UnknownSPMValidationError extends BaseSPMError {
  public name = 'Unknown Snapshot Pull Model Error';
  public constructor(shouldContactCoveo: boolean) {
    super(
      "An unknown error occured while validating the custom template. Try recreating it from 'empty' or 'full'.",
      shouldContactCoveo
    );
  }
}

export class InvalidSPMError extends BaseSPMError {
  public name = 'Invalid Snapshot Pull Model Error';
  public constructor(shouldContactCoveo: boolean, errors: ValidationError[]) {
    super(getPrettyJsonValidationErrors(errors), shouldContactCoveo);
  }
}
