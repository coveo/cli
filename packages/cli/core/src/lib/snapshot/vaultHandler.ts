import {
  VaultEntryModel,
  VaultValueType,
  VaultVisibilityType,
} from '@coveord/platform-client';
import {CliUx} from '@oclif/core';
import {readJsonSync, rmSync, writeJsonSync} from 'fs-extra';
import open from 'open';
import {join} from 'path';
import {cwd} from 'process';
import {green, inverse} from 'chalk';
import {
  InvalidVaultEntryError,
  InvalidVaultFileError,
  MissingVaultEntryValueError,
} from '../errors/vaultErrors';
import {AuthenticatedClient} from '@coveo/cli-commons/src/platform/authenticatedClient';
import {Snapshot} from './snapshot';
import {VaultEntryAttributes} from './snapshotReporter';
import {Plurable, pluralizeIfNeeded} from '@coveo/cli-commons/src/utils/string';
import {ProcessAbort} from '../../lib/errors/processError';
import dedent from 'ts-dedent';

export class VaultHandler {
  private static readonly defaultEntryValue = '';

  public constructor(private organizationId: string) {}

  public async createEntries(entries: VaultEntryAttributes[]) {
    await this.prepareAndOpenFile(entries);
    await this.ensureValidityOrAbort(entries);
    const vaultEntryModels = this.getVaultEntryModels(entries);
    rmSync(this.vaultEntryFilePath);
    await this.doCreateVaultEntries(vaultEntryModels);
  }

  private async prepareAndOpenFile(entries: VaultEntryAttributes[]) {
    this.prepareFile(entries);
    await this.openFile();
  }

  private async ensureValidityOrAbort(entries: VaultEntryAttributes[]) {
    let valid = false;
    while (!valid) {
      const key = await CliUx.ux.prompt('', {
        type: 'single',
        required: false,
        prompt: `\n${inverse('Press any key to continue. Press q to abort')}\n`,
      });
      if (key === 'q') {
        throw new ProcessAbort();
      }
      try {
        this.ensureEntriesValidity(entries);
        valid = true;
      } catch (error) {
        CliUx.ux.log('');
        CliUx.ux.warn(`${error}`);
        await this.openFile(false);
      }
    }
  }

  private async doCreateVaultEntries(vaultEntryModels: VaultEntryModel[]) {
    const client = await this.client();
    if (vaultEntryModels.length > 0) {
      const entryPlurable: Plurable = ['entry', 'entries'];
      CliUx.ux.action.start(
        `Creating vault ${pluralizeIfNeeded(
          entryPlurable,
          vaultEntryModels.length
        )}`
      );
      await Promise.all(
        vaultEntryModels.map((entry) => client.vault.create(entry))
      );
      CliUx.ux.action.stop(green('âœ”'));
    }
  }

  private prepareFile(entries: VaultEntryAttributes[]) {
    const data: Record<string, unknown> = {};
    for (const {vaultEntryId} of entries) {
      data[vaultEntryId] = VaultHandler.defaultEntryValue;
    }

    writeJsonSync(this.vaultEntryFilePath, data, {spaces: 4});
  }

  private async openFile(print = true) {
    if (print) {
      CliUx.ux.log(dedent`\nOpening file ${this.vaultEntryFilePath}.
      Please fill out all the vault entries from the JSON file.`);
    }
    await open(this.vaultEntryFilePath);
  }

  private ensureEntriesValidity(
    requiredEntries: VaultEntryAttributes[]
  ): void | never {
    const missingEntries: string[] = [];
    let data: Record<string, unknown>;

    try {
      data = readJsonSync(this.vaultEntryFilePath);
    } catch (error) {
      throw new InvalidVaultFileError(error);
    }

    for (const {vaultEntryId} of requiredEntries) {
      if (
        data[vaultEntryId] === undefined ||
        data[vaultEntryId] === VaultHandler.defaultEntryValue
      ) {
        missingEntries.push(vaultEntryId);
      }
    }
    if (missingEntries.length > 0) {
      throw new MissingVaultEntryValueError(missingEntries);
    }
  }

  private getVaultEntryModels(
    entries: VaultEntryAttributes[]
  ): VaultEntryModel[] {
    const data = readJsonSync(this.vaultEntryFilePath);
    const models: VaultEntryModel[] = [];
    for (const {resourceName, resourceType, vaultEntryId} of entries) {
      const jsonPath = this.getVaultEntryJsonPath(vaultEntryId, resourceName);
      models.push({
        key: vaultEntryId,
        attributeReferences: [
          {
            jsonPath,
            resourceName,
            resourceType,
          },
        ],
        organizationId: this.organizationId,
        value: data[vaultEntryId],
        valueType: VaultValueType.STRING,
        vaultVisibilityType: VaultVisibilityType.STRICT,
      });
    }
    return models;
  }

  private getVaultEntryJsonPath(key: string, resourceName: string) {
    // TODO: CDX-939: Define contract with backend for report and upcoming contract.
    // The report should give the vault entry JSONpath in the same format we need to send it
    const regex = `${resourceName}-(?<jsonPath>.*)`;
    const match = key.match(regex);
    const jsonPath = match?.groups?.['jsonPath'];
    if (!jsonPath) {
      throw new InvalidVaultEntryError(this.organizationId, key);
    }
    return jsonPath;
  }

  private async client(organization = this.organizationId) {
    return await new AuthenticatedClient().getClient({organization});
  }

  private get vaultEntryFilePath() {
    return join(cwd(), `${this.organizationId}-vault.json`);
  }
}
