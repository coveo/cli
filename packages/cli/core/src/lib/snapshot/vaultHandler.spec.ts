jest.mock('open');
jest.mock('fs-extra');
jest.mock('@coveo/cli-commons/platform/authenticatedClient');

import {ResourceSnapshotType} from '@coveord/platform-client';
import {VaultEntryAttributes} from './snapshotReporter';
import {VaultHandler} from './vaultHandler';
import {AuthenticatedClient} from '@coveo/cli-commons/platform/authenticatedClient';
import {readJsonSync, rmSync, writeJsonSync} from 'fs-extra';
import open from 'open';
import {CliUx} from '@oclif/core';
import {ProcessAbort} from '../errors/processError';

const mockedAuthenticatedClient = jest.mocked(AuthenticatedClient);
const mockedCreate = jest.fn();
const mockedPrompt = jest.fn();
const mockedWriteJsonSync = jest.mocked(writeJsonSync);
const mockedReadJsonSync = jest.mocked(readJsonSync);
const mockedRmSync = jest.mocked(rmSync);
const mockedOpen = jest.mocked(open);
const mockedCliUx = jest.spyOn(CliUx.ux, 'warn');

const doMockPrompt = () => {
  Object.defineProperty(CliUx.ux, 'prompt', {value: mockedPrompt});
};
const doMockAuthenticatedClient = () => {
  mockedAuthenticatedClient.mockImplementation(
    () =>
      ({
        getClient: () => Promise.resolve({vault: {create: mockedCreate}}),
      } as unknown as AuthenticatedClient)
  );
};

describe('VaultHandler', () => {
  const vaultEntryFile = 'myorgid-vault.json';
  const vault = new VaultHandler('myorgid');
  const anyKey = 'a';
  const abortKey = 'q';
  const vaultEntries: VaultEntryAttributes[] = [
    {
      vaultEntryId: 'sitemap_yyyyyy-configuration.json.path',
      resourceName: 'sitemap_yyyyyy',
      resourceType: ResourceSnapshotType.source,
    },
    {
      vaultEntryId: 'web_xxxxxx-configuration.json.path',
      resourceName: 'web_xxxxxx',
      resourceType: ResourceSnapshotType.source,
    },
  ];

  beforeAll(() => {
    doMockPrompt();
    doMockAuthenticatedClient();
  });

  afterAll(() => {
    mockedPrompt.mockReset();
    mockedReadJsonSync.mockReset();
  });

  describe('when the vault entries are valid', () => {
    beforeAll(() => {
      mockedReadJsonSync.mockReturnValue({
        'sitemap_yyyyyy-configuration.json.path': 'secret',
        'web_xxxxxx-configuration.json.path': 'another-secret',
      });
      mockedPrompt.mockReturnValue(anyKey);
    });

    beforeEach(async () => {
      await vault.createEntries(vaultEntries);
    });

    it('#createEntries should create a file with missing vault entries', () => {
      expect(mockedWriteJsonSync).toHaveBeenCalledWith(
        expect.stringContaining(vaultEntryFile),
        {
          'sitemap_yyyyyy-configuration.json.path': '',
          'web_xxxxxx-configuration.json.path': '',
        },
        {spaces: 4}
      );
    });

    it('#createEntries should open the right file', () => {
      expect(mockedOpen).toHaveBeenCalledWith(
        expect.stringContaining(vaultEntryFile)
      );
    });

    it('#createEntries should prompt instructions for the user', () => {
      expect(mockedPrompt).toBeCalledTimes(1);
      expect(mockedPrompt.mock.lastCall).toMatchSnapshot();
    });

    it('#createEntries should send as many requests as there is entries to create', () => {
      expect(mockedCreate).toHaveBeenCalledTimes(2);
    });

    it('#createEntries should delete file', () => {
      expect(mockedRmSync).toBeCalledWith(
        expect.stringContaining(vaultEntryFile)
      );
    });

    it.each([
      {
        idx: 1,
        jsonPath: 'configuration.json.path',
        resourceName: 'sitemap_yyyyyy',
        key: 'sitemap_yyyyyy-configuration.json.path',
        value: 'secret',
      },
      {
        idx: 2,
        jsonPath: 'configuration.json.path',
        resourceName: 'web_xxxxxx',
        key: 'web_xxxxxx-configuration.json.path',
        value: 'another-secret',
      },
    ])(
      '#createEntries should send requests to create vault entries',
      ({idx, jsonPath, key, resourceName, value}) => {
        expect(mockedCreate).toHaveBeenNthCalledWith(idx, {
          attributeReferences: [
            {
              jsonPath,
              resourceName,
              resourceType: 'SOURCE',
            },
          ],
          key,
          organizationId: 'myorgid',
          value,
          valueType: 'STRING',
          vaultVisibilityType: 'STRICT',
        });
      }
    );
  });

  describe('when the user does not fill all the vault entries values', () => {
    beforeAll(() => {
      mockedReadJsonSync
        .mockReturnValueOnce({
          'sitemap_yyyyyy-configuration.json.path': '',
          'web_xxxxxx-configuration.json.path': '',
        })
        .mockReturnValueOnce({
          'sitemap_yyyyyy-configuration.json.path': 'secret',
          'web_xxxxxx-configuration.json.path': 'another-secret',
        });
      mockedPrompt.mockReturnValue(anyKey);
    });

    beforeEach(async () => {
      await vault.createEntries(vaultEntries);
    });

    it('#createEntries should warn the user about missing vault entries', () => {
      expect(mockedCliUx).toMatchSnapshot();
    });
  });

  describe('when the user refuses to create missing vault entries', () => {
    beforeAll(() => {
      mockedPrompt.mockReturnValue(abortKey);
    });

    it('#createEntries should not create vault entries', async () => {
      await expect(() => vault.createEntries(vaultEntries)).rejects.toThrow(
        new ProcessAbort()
      );
    });
  });
});
