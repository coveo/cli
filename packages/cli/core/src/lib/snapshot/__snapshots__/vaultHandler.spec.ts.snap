// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaultHandler when the user does not fill all the vault entries values #createEntries should warn the user about missing vault entries 1`] = `
[MockFunction] {
  "calls": [
    [
      "Missing vault entry value: 
  â€¢ sitemap_yyyyyy-configuration.json.path
  â€¢ web_xxxxxx-configuration.json.path

Fill all required vault entries to proceed.",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`VaultHandler when the vault entries are valid #createEntries should prompt instructions for the user 1`] = `
[
  "",
  {
    "prompt": "
[7mPress any key to continue. Press q to abort[27m
",
    "required": false,
    "type": "single",
  },
]
`;
